src/librustc/middle/traits/mod.rs

```diff
@@ -220,6 +220,9 @@ pub enum Vtable<'tcx, N> {
     /// Vtable identifying a particular impl.
     VtableImpl(VtableImplData<'tcx, N>),
 
+    /// Vtable for default trait implementations
```

Comment:
Let's expand this comment a bit more. I think this corresponds to what happens when a `impl Foo for ..` impl is matched, right? I also suspect it should carry some nested obligations, which would will correspond to the supertraits of the defaulted trait.